### PR TITLE
dev-org/slides: Fill contents

### PR DESCRIPTION
**Pull request by razvand**
# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/open-education-hub/ccas-internal/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [n/a] Tested your changes against relevant architectures and platforms;
- [n/a] Updated relevant documentation (if needed).

## Description of changes

Fill contents for slides section. Add information on storing, writing, generating and delivering slides.

### CODE REVIEW

File: chapters/develop-organize/slides/reading/read.md [hunk 1]
```diff
@@ -22,58 +22,215 @@ Our choice of engine for storing and rendering slides is [`reveal-md`](https://g
...
 
-2D view of slides for `markdown-pp`
+An index file, as above, would consist of:
 
-keyboard shortcuts
+- references to slide sections, as in the line `[The Software Stack](#software-stack)`
+- inclusion of actual slide contents files, as in the line `!INCLUDE "overview/slides/software-stack.md"`
 
-## Slide Contents
+Each actual content slide Markdown file will feature information related to the topic, using the features and syntax of `reveal-md` / `reveal.js`.
```
Comment (teodutu) [inline chapters/develop-organize/slides/reading/read.md:146]:
```suggestion
Each actual content slide Markdown file will contain information related to the topic, using the features and syntax of `reveal-md` / `reveal.js`.
```
avoid repeating "feature"

File: chapters/develop-organize/slides/reading/read.md [hunk 2]
```diff
@@ -22,58 +22,215 @@ Our choice of engine for storing and rendering slides is [`reveal-md`](https://g
...
+Depending on the content, there are some common type of slides:
```
Comment (teodutu) [inline chapters/develop-organize/slides/reading/read.md:147]:
```suggestion
Depending on the content, there are some common types of slides:
```

File: chapters/develop-organize/slides/reading/read.md [hunk 3]
```diff
@@ -22,58 +22,215 @@ Our choice of engine for storing and rendering slides is [`reveal-md`](https://g
...
+
+- text slides: consisting only / mostly of text items, that may be structured as bullets
+- title slides: consisting only of a title, or one line, to introduce a new topic
+- code snippet slides: consisting of sequence of code to demonstrate a concept
+- media slides: consisting of images, videos or audio items
+- table slides: consisting of a table or a comparison of ideas
```
Comment (teodutu) [inline chapters/develop-organize/slides/reading/read.md:153]: You can also combine these to a certain extent, such as having an image and 2-3 bullets. I wouldn't categorise them so granularly. Maybe rephrase these as types of content that can be present on slides.

File: chapters/develop-organize/slides/reading/read.md [hunk 4]
```diff
@@ -22,58 +22,215 @@ Our choice of engine for storing and rendering slides is [`reveal-md`](https://g
...
+1. generate HTML output
+1. generate PDF output
+
+These steps are generally encoded in the `Makefile`.
+
+To preprocess slides, we use `markdownpp`.
+The typical command for doing this is:
+
+```console
+markdownpp slides.mdpp -o slides.md
```
Comment (teodutu) [inline chapters/develop-organize/slides/reading/read.md:187]:
```suggestion
$ markdownpp slides.mdpp -o slides.md
```

File: chapters/develop-organize/slides/reading/read.md [hunk 5]
```diff
@@ -22,58 +22,215 @@ Our choice of engine for storing and rendering slides is [`reveal-md`](https://g
...
+```
+
+This processes the `slides.mdpp` file and generates the `slides.md` file.
+
+To generate a static HTML output, use `reveal-md`:
+
+```console
+reveal-md slides.md --static _site/
```
Comment (teodutu) [inline chapters/develop-organize/slides/reading/read.md:195]:
```suggestion
$ reveal-md slides.md --static _site/
```

File: chapters/develop-organize/slides/reading/read.md [hunk 6]
```diff
@@ -22,58 +22,215 @@ Our choice of engine for storing and rendering slides is [`reveal-md`](https://g
...
+
+```console
+$ reveal-md slides.md
+Reveal-server started at http://localhost:1948
+The slides are at http://localhost:1948/slides.md
+```
 
-index slide
+This is required for generating a PDF output.
+We use [`decktape`](https://github.com/astefanutti/decktape) to generate PDF slides;
```
Comment (teodutu) [inline chapters/develop-organize/slides/reading/read.md:210]: TIL

**Review by teodutu — CHANGES_REQUESTED** [review]

File: chapters/develop-organize/slides/reading/read.md (hunk 1 above)
Comment (teodutu) [reply chapters/develop-organize/slides/reading/read.md:146]: Bump on this.

**Review by teodutu — APPROVED** [review]
Except for repeating the word "feature", things look alright.

**Comment by razvand** [comment]
> Except for repeating the word "feature", things look alright.

fixed
